### PR TITLE
Use tsurlfilter's DNS engine

### PR DESCRIPTION
**Pull request by mjethani**
This patch uses the DNS engine in @adguard/tsurlfilter for hosts-only mode.

### CODE REVIEW

**Comment by mjethani** [comment]
@ameshkov FYI

Hosts-only mode is for testing blocking at the host level only. The `DnsEngine` object is perfect for this.

In actual DNS mode there would be no URL from which to extract the hostname, but the `extractHostname()` function is extremely fast and I don't think it makes any significant difference to the benchmark results.

File: packages/adblocker-benchmarks/blockers/tsurlfilter.js [hunk 1]
```diff
@@ -39,18 +46,33 @@ module.exports = class TSUrlFilter {
...
 
   constructor(engine) {
     this.engine = engine;
   }
 
   match({ url, frameUrl, type }) {
+    const result = TSUrlFilter.hostsOnly ?
+                     this.matchHostname(url) :
+                     this.matchRequest({ url, frameUrl, type });
+    return result !== null && !result.whitelist;
```
Comment (mjethani) [inline packages/adblocker-benchmarks/blockers/tsurlfilter.js:63]: Something worth noting is that the DNS engine also supports whitelisting rules. This is also true of other content blockers like uBlock Origin. But we have no whitelisting rules in our [`hosts.txt`](https://github.com/cliqz-oss/adblocker/blob/master/packages/adblocker-benchmarks/hosts.txt) file here.

**Review by remusao — APPROVED** [review]

File: packages/adblocker-benchmarks/blockers/tsurlfilter.js (hunk 1 above)
Comment (ameshkov) [reply packages/adblocker-benchmarks/blockers/tsurlfilter.js:63]: Well, it also supports a little different subset of rules & modifiers, the syntax is explained here:
https://github.com/AdguardTeam/AdGuardHome/wiki/Hosts-Blocklists

**Review by ameshkov — APPROVED** [review]

**Comment by mjethani** [comment]
@ameshkov thanks for helping with the integration.

The only thing left now is serialization and deserialization. I could not find anything in the `README.md` file for this. I see there's a `RuleStorage` object, but I'm not sure how to use it.

Since the library is under development, it's not so important to implement this now.

In the future, when an API is available, we'll have to implement the following two functions in `blockers/tsurlfilter.js` in this repo:

```js
class TSUrlFilter {
  async serialize() {
    // return data in serialized format
  }

  async deserialize(serialized) {
    // initialize engine with given data in serialized format
  }
}
```

**Pull request merged**